src/plugins/avoidance/avoidance.cpp

```diff
@@ -0,0 +1,27 @@
+#include "plugins/avoidance/avoidance.h"
+#include "avoidance_impl.h"
+#include <vector>
+
+namespace mavsdk {
+
+Avoidance::Avoidance(System& system) : PluginBase(), _impl{new AvoidanceImpl(system)} {}
+
+Avoidance::~Avoidance() {}
+
+void Avoidance::trajectory_representation_waypoints_async(
+    trajectory_representation_waypoints_callback_t callback)
+{
+    return _impl->trajectory_representation_waypoints_async(callback);
+}
+
+const char* Avoidance::result_str(Result result)
```

Comment:
This `Result` is not used anywhere. This either needs to be implemented and used or removed.